// Messages between main and media server
syntax = "proto3";

package org.couchers.media;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

message UploadRequest {
  // This service is called only by the backend to coordinate file upload between the backend and media servers

  enum UploadType {
    IMAGE = 0;
  }

  string key = 1;
  UploadType type = 2;

  google.protobuf.Timestamp created = 3;
  google.protobuf.Timestamp expiry = 4;

  uint32 max_width = 5;
  uint32 max_height = 6;
}

service Media {
  rpc UploadConfirmation(UploadConfirmationReq) returns (google.protobuf.Empty) {
    // when an upload completes, the upload server confirms to the main server
    // that the upload succeeded using this function
  }
}

message UploadConfirmationReq {
  string key = 1;
  string filename = 2;
}
